### PR TITLE
Feature/analyses precip stats

### PR DESCRIPTION
**Pull request by PerryShafran-NOAA**
## Pull Request Testing ##

This PR is for adding in the stats scripts for precipitation analyses (rtma, urma, ccpa).  There is no prep for precip analyses.

- [ ] Describe testing already performed for this Pull Request:</br>

Daily testing of the scripts, running at each hour.  

- [ ] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

After bringing in the scripts and adding in the fix directory, it is suggested that this script be run hourly, as in grid2obs precipitation.  Use -v $vhr every time you submit the job, like this:

```
qsub -v vhr=12 jevs_analyses_urma_precip_stats.sh
qsub -v vhr=12 jevs_analyses_rtma_precip_stats.sh
qsub -v vhr=12 jevs_analyses_ccpa_precip_stats.sh
```

The need to run the jobs hourly is because the CoCoRaHS observational dataset is to produce a set of observations at 7 AM LST, which means it's a different UTC hour depending on location.  Most of the obs congregate around 12Z and 13Z.  All observations are 24-hour precip totals.  

When you run with vhr=23, that job also runs the gather StatAnalysis job.  Be sure that there are no other precip analysis jobs running when you run the vhr=23 step.  

- [ ] Has the code been checked to ensure that no errors occur during the execution? **[Yes or No]**

Yes

- [ ] Do these updates/additions include sufficient testing updates? **[Yes or No]**

Yes

- [ ] Please complete this pull request review by **[Fill in date]**.</br>

When you can.  

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

**Comment by ShelleyMelchior-NOAA** [comment]
Instructions seem straight forward. I will clone your fork and checkout `feature/analyses_precip` soon.

Do you plan to include the ecf scripts in this PR, or will they be included later?

**Comment by PerryShafran-NOAA** [comment]
I hadn't thought about ecf files, but I can easily add them before the PR is approved.

**Comment by AliciaBentley-NOAA** [comment]
Thanks, @PerryShafran-NOAA! I think that would be a great thing to add to this PR before it is merged.

**Comment by PerryShafran-NOAA** [comment]
I'll work on that.

File: dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh]: @PerryShafran-NOAA , can you update your drivers to revert HOMEevs back to the default value? I will make the change in my local checkout, but we need it changed before merging.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh]: Yes, done.  I can't believe that I forgot to remove the testing directory.

**Comment by ShelleyMelchior-NOAA** [comment]
I ran a single run (vhr=0) for ccpa precip stats and produced a small stat file. It is tiny, but maybe that is OK after reading your note that most obs congregate around 12Z and 13Z. I re-ran the test of ccpa for vhr=12. Can you confirm the small stat file looks as expected before I proceed w/ running all of the other vhr's?
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_ccpa_precip_stats.o154170172`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses/atmos.20240624/ccpa/precip`

**Comment by PerryShafran-NOAA** [comment]
I will look.  Yes, tiny is to be expected.  There should be only 3 lines in the small stat file.

**Comment by PerryShafran-NOAA** [comment]
Looks good.  And for 00Z, there shouldn't be any observations.  I may consider running this on fewer hours of the day, but for now, hourly runs are what I came down on, as I don't know which hours appear in the CoCoRaHS files until after I run stats.

**Comment by AliciaBentley-NOAA** [comment]
@ShelleyMelchior-NOAA @PerryShafran-NOAA Just another quick comment related to adding the .ecf scripts before this PR is merged: Please also make updates to the [ecf/defs/evs-nco.def](https://github.com/NOAA-EMC/EVS/blob/develop/ecf/defs/evs-nco.def) file when you add the new .ecf scripts. Thank you!

**Comment by ShelleyMelchior-NOAA** [comment]
The test for ccpa stats is complete! All vhr's have been run. Please review.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_ccpa_precip_stats.o15417*` with a timestamp after Jun 27 19:00.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

**Comment by PerryShafran-NOAA** [comment]
Oh, I just realized that I need to add EVSINccpa to the J-job.  Since I have that set in the dev driver, it should be set in the J-job as well.  Stay tuned for that.

**Comment by AliciaBentley-NOAA** [comment]
@PerryShafran-NOAA Since you're making some changes, can you also update the top of [scripts/stats/analyses/exevs_analyses_rtma_precip_stats.sh](https://github.com/NOAA-EMC/EVS/pull/492/files#diff-ecc62c1d13b44bbbd1f5e58412a95e88024527f67025edf6ebc89dbd962f8998) match the top of the other two exevs_analyses_*_precip_stats.sh scripts? The other two look the same, and this one looks different. Thanks! 

CC @ShelleyMelchior-NOAA

**Comment by ShelleyMelchior-NOAA** [comment]
Ok, I have re-run **_ccpa_** precip for all vhr's, but this time for VDATE=20240625. Please review.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_ccpa_precip_stats.o154280*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

In the interim, I will do the same processing for rtma and urma precip stats.

**Comment by ShelleyMelchior-NOAA** [comment]
I have run **_rtma_** precip for all vhr's for VDATE=20240625. Please review.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_rtma_precip_stats.o1542833*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

I have run **_urma_** precip for all vhr's for VDATE=20240625. Please review.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_urma_precip_stats.o154284*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

**Comment by PerryShafran-NOAA** [comment]
In the ccpa scripts:

`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/EVS/jobs/JEVS_ANALYSES_STATS: line 79: EVSINccpa: /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/prep/cam: syntax error: operand expected (error token is "/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/prep/cam")`

Something is amiss in the EVSINccpa definition in the J-job.  I just fixed that and the ccpa should be run again.

The rtma output is fine.

**Comment by ShelleyMelchior-NOAA** [comment]
Great. I will pull in the change, and re-run ccpa.

rtma is good. ✔️ 

Did you check urma? I provided urma results along w/ rtma results.

**Comment by PerryShafran-NOAA** [comment]
Yes, urma is fine.  

In addition, it would also good to do additional tests for dates when either the cocorahs file and/or the analyses are not there, to test the missing data capabilities.

**Comment by ShelleyMelchior-NOAA** [comment]
Can do! I will re-run ccpa. And I will also do a missing data run for each analysis using VDATE=20240621.

**Comment by ShelleyMelchior-NOAA** [comment]
Ok, I have re-run ccpa precip for all vhr's, but this time for VDATE=20240628. Please review.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_ccpa_precip_stats.o15458*` (timestamp Jul 1)
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

**Comment by PerryShafran-NOAA** [comment]
The ccpa run checks out.

**Comment by ShelleyMelchior-NOAA** [comment]
I tested runs of ccpa, rtma, and urma for VDATE=20240621. Email recipients confirmed receipt of missing data emails.

**Comment by PerryShafran-NOAA** [comment]
There were missing data emails, WARNINGs of missing data in the .o files, and no FATAL ERRORs.  I think we are all good on this.

**Comment by ShelleyMelchior-NOAA** [comment]
Great! Let me do a closer code review. My initial glance was quick.

File: dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh
Comment (AliciaBentley-NOAA) [inline dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh:37]: @PerryShafran-NOAA @ShelleyMelchior-NOAA Do we want these /evs_test directories to make their way into develop in the /dev/drivers?

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh:37]: They've been that way since forever and that's how it's being run for all of EVS in emc.vpppg.

Comment (AliciaBentley-NOAA) [reply dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh:37]: @PerryShafran-NOAA Ok! As long as this isn't new and follows everything else! Thanks.

Comment (PerryShafran-NOAA) [reply dev/drivers/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats.sh:37]: Yup.  I started with the grid2obs driver scripts when I made the precip scripts and just made modifications for precip.  It's definitely not new.

File: parm/metplus_config/stats/analyses/precip/PointStat_fcstCCPA_obsCOCORAHS_ASCIIprecip.conf
Comment (AliciaBentley-NOAA) [inline parm/metplus_config/stats/analyses/precip/PointStat_fcstCCPA_obsCOCORAHS_ASCIIprecip.conf]: @PerryShafran-NOAA I think we removed all URLs from the ops EVS v1.0 code delivery? Are these https lines new? Can you please remove them? We'll also need to remove any other web links like this that got introduced between then and now. 

CC @ShelleyMelchior-NOAA

Comment (PerryShafran-NOAA) [reply parm/metplus_config/stats/analyses/precip/PointStat_fcstCCPA_obsCOCORAHS_ASCIIprecip.conf]: I removed the DTC websites from the new precip parm files.  They were there because I got these parm files from DTC, but they are removed now.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Don't merge yet; I need to make few corrections regarding the mask directories so that they are what they should be.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Modified the masks directories in parm and ex scripts to use the $MASKS directory variable.  I ask that you re-run ONE script (valid vhr=12) from each model to make this test.

**Comment by ShelleyMelchior-NOAA** [comment]
Sure, Perry. I had not yet started the official code review. I will do the test re-runs as specified and update this conversation once complete. I will commence the code review to follow.

**Comment by ShelleyMelchior-NOAA** [comment]
I pulled in the changes and I re-ran all 3 precip drivers for vhr=12 and VDATE=20240623. The COMOUTsmall directories are empty. I see FATAL ERROR in the log files. Maybe I didn't set something up correctly ...

logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/ jevs_analyses_*a_precip_stats.o1546006*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

**Comment by PerryShafran-NOAA** [comment]
It could be real - like I set up the variable incorrectly in either the ex script or the parm file.  Let me see what the issue is.

**Comment by PerryShafran-NOAA** [comment]
```
FileNotFoundError: [Errno 2] No such file or directory: '/lfs/h1/ops/dev/dcom/20240623/validation_data/CoCoRaHS/cocorahs.20240
623.dailyprecip.csv'
```

Why is it looking for /lfs/h1/ops/dev/dcom?  Let me look if this is something I accidentally changed, but I only changed mask stuff, not the location of the cocorahs file.

**Comment by PerryShafran-NOAA** [comment]
OK, I see the problem.  Fix is forthcoming.

**Comment by PerryShafran-NOAA** [comment]
OK, fixes made, please try again.  

Also, you may consider returning VDATE to the current usage.

**Comment by ShelleyMelchior-NOAA** [comment]
Thanks for the fix, Perry. I cleared out the COMOUT dirs for VDATE=20240623 and ran again for vhr=12. I kept w/ 20240623 b/c I wanted a VDATE that hadn't been checked out yet.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_analyses_*a_precip_stats.o1546067*`
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v2.0/stats/analyses`

File: ecf/defs/evs-nco.def
Comment (ShelleyMelchior-NOAA) [inline ecf/defs/evs-nco.def]: Please check line 963. I think it needs to end in `_vhr_11`

File: ecf/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats_vhr_master.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats_vhr_master.ecf]: Check line 1. I think it should be `_precip_` instead of `_grid2obs_`

File: ecf/scripts/stats/analyses/jevs_analyses_rtma_precip_stats_vhr_master.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/stats/analyses/jevs_analyses_rtma_precip_stats_vhr_master.ecf]: Check line 1. I think it should be `_precip_` instead of `_grid2obs_`

File: ecf/scripts/stats/analyses/jevs_analyses_urma_precip_stats_vhr_master.ecf
Comment (ShelleyMelchior-NOAA) [inline ecf/scripts/stats/analyses/jevs_analyses_urma_precip_stats_vhr_master.ecf]: Check line 1. I think it should be `_precip_` instead of `_grid2obs_`

File: ecf/defs/evs-nco.def
Comment (PerryShafran-NOAA) [reply ecf/defs/evs-nco.def]: Fixed.

File: ecf/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats_vhr_master.ecf
Comment (PerryShafran-NOAA) [reply ecf/scripts/stats/analyses/jevs_analyses_ccpa_precip_stats_vhr_master.ecf]: Fixed.

File: ecf/scripts/stats/analyses/jevs_analyses_rtma_precip_stats_vhr_master.ecf
Comment (PerryShafran-NOAA) [reply ecf/scripts/stats/analyses/jevs_analyses_rtma_precip_stats_vhr_master.ecf]: Fixed.

File: ecf/scripts/stats/analyses/jevs_analyses_urma_precip_stats_vhr_master.ecf
Comment (PerryShafran-NOAA) [reply ecf/scripts/stats/analyses/jevs_analyses_urma_precip_stats_vhr_master.ecf]: Fixed.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
Code changes and new code look good. This PR is approved for merge.

**Comment by PerryShafran-NOAA** [comment]
@ShelleyMelchior-NOAA Ah, sorry.  I did check those 12Z files earlier, and they looked good, but I guess that I forgot to report that here.  Consider those validated.

**Comment by ShelleyMelchior-NOAA** [comment]
This PR is approved for merge.

**Pull request merged**
